db/migrate/20180410071554_current_policy.rb

```diff
@@ -0,0 +1,58 @@
+Sequel.migration do
+  up do
+    alter_table :policy_versions do
+      set_column_type :created_at, :timestamptz
+      add_column :commited_at, :timestamptz,
+        index: true, # to quickly find the null
+        null: true # will be enforced by a trigger
+      add_constraint(:created_before_commit) { created_at <= commited_at }
+    end
+
+    execute """
+      CREATE OR REPLACE FUNCTION policy_versions_commit()
+        RETURNS trigger
+      LANGUAGE plpgsql AS $$
+        BEGIN
+          UPDATE policy_versions pv
+            SET commited_at = clock_timestamp()
+            WHERE commited_at IS NULL;
+          RETURN new;
+        END;
+      $$;
+
+      -- deferred constraint trigger will run on transaction commit
```

Comment:
But we are calling `emit_audit` before the transaction is committed, so `policy_versions_commit` will not have happened. Yes?